lib/trino-hive-formats/src/main/java/io/trino/hive/formats/line/grok/Discovery.java

```diff
@@ -0,0 +1,196 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.hive.formats.line.grok;
+
+import io.trino.hive.formats.line.grok.exception.GrokException;
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.Iterator;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.TreeMap;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * {@code Discovery} try to find the best pattern for the given string.
+ *
+ * @author anthonycorbacho
+ * @since 0.0.2
+ */
+// Note: this code is forked from oi.thekraken.grok.api
+// Copyright 2014 Anthony Corbacho, and contributors.
+public class Discovery
+{
+    private Grok grok;
+
+    /**
+     * Create a new {@code Discovery} object.
+     *
+     * @param grok instance of grok
+     */
+    public Discovery(Grok grok)
+    {
+        this.grok = grok;
+    }
+
+    /**
+     * Sort by regex complexity.
+     *
+     * @param groks Map of the pattern name and grok instance
+     * @return the map sorted by grok pattern complexity
+     */
+    private Map<String, Grok> sort(Map<String, Grok> groks)
+    {
+        List<Grok> groky = new ArrayList<Grok>(groks.values());
+        Map<String, Grok> mGrok = new LinkedHashMap<String, Grok>();
```

Comment:
please find any variable named like this and switch it to modern naming conventions used in Trino :)